src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationNotifications.cs

```diff
@@ -44,7 +46,19 @@ public OpenIdConnectAuthenticationNotifications()
         /// <summary>
         /// Invoked to manipulate redirects to the identity provider for SignIn, SignOut, or Challenge.
         /// </summary>
-        public Func<RedirectToIdentityProviderNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions>, Task> RedirectToIdentityProvider { get; set; }
+        public Func<RedirectToIdentityProviderNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions>, Task> RedirectToIdentityProvider
```

Comment:
This can work in this case, but for the notifications themselves, since the Ticket and Message can be set, we need some stronger validation.

I noticed in FB and Goodle MW and Handlers we don't check for null in public methods.